packages/material-ui/src/Menu/Menu.js

```diff
@@ -131,10 +131,17 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     if (event.key === 'ArrowLeft' && isSubMenu) {
       // This assigns the focusVisible class to the parent item when closing
       // a sub menu using the left arrow keyboard navigation.
-      anchorEl.onfocus = async () => {
-        await anchorEl.classList.remove('MuiMenuItem-openSubMenuParent');
-        anchorEl.classList.add('Mui-focusVisible');
-      };
+      
+      // TODO: @eps1lon we were just doing this before the change to `useIsFocusVisible` last August.
+      // TODO: It would be nice to figure out why this stopped working after that change.
+      anchorEl.focus();
```

Comment:
Is the `anchorEl` accessible at this point? If we could entangle the generic Menu from the SubMenu it would be easier to change behavior in one without affecting the other. I suspect that this wants to return focus but either the `anchorEl` is not focusable at this point or the focus trap is still active.

I suggest checking out what ends up getting focused and if there are intermediate focus/blur events (and how their timings are).

Reply:
The parent item does get focused. It just doesn't get the `focusVisible` classs that makes it look focused.

Reply:
Hence, all the shenanigans we were trying with dom manip and such.

Reply:
And `anchorEl` is accessible here and is just the parent item the submenu is attached to. Also the handler that this is happening in will ONLY fire on a submenu, so there is no danger of regular menu behavior changing.